feat: Add column statistics indicator for columns with TableColumnStats available

mgorsk1 opened the pull request:
### Summary of Changes

Currently there is no indication if there is a need to expand the column details to see whether column has column stats available or not. Users need to expand details even if there is nothing there.

This PR introduces an icon which indicates that a given column has statistics available. I am not strongly opinionated towards icon or placement choice but I think such indicator is a must to improve user experience.

This is how it looks like after this initial implementation:
![image](https://user-images.githubusercontent.com/5680655/151964237-715d0c39-9f3c-46c1-ac44-7d45d35ce35a.png)

### Tests

tbd

### Documentation

tbd

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes.
- [] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

mgorsk1 commented:
@Golodhros @danwom I'd appreciate any feedback you might have on this 🙏

mgorsk1 commented:
ping @Golodhros

Golodhros commented:
Pinging Knowl!

knowlb commented:
We love the idea of indicating whether there are column stats to expand, but the faint icon on the far right was really hard to notice, so would like to suggest using this treatment (see attached img):
![column stats indicator](https://user-images.githubusercontent.com/38537196/153060674-f174b780-c07b-4e69-8507-81023879e180.png)
This is also to be consistent with a new feature we're developing to highlight search keyword matches in results, including pulling out column names for tables and charts & queries for dashboards: 
<img width="603" alt="image" src="https://user-images.githubusercontent.com/38537196/153061134-7db30d10-5c2c-45f3-98e3-e84dacffaac1.png">

mgorsk1 commented:
thanks @knowlb that totally makes sense! where do you source your icons from (or are they custom made) so I can submit PR with matching aesthetics?

mgorsk1 commented:
I refactored to your suggestion @knowlb with minor adjustment - I didn't like the fact that icon presence could shift some column names and in results they'd not be aligned throughout so I've added the icon to the left of col name & description. Let me know wdyt

![Uploading image.png…]()

knowlb commented:
Thanks and like the alignment idea too! Looks great-

Golodhros commented on frontend/amundsen_application/static/js/features/ColumnList/index.tsx:
```diff
@@ -309,6 +311,18 @@ const ColumnList: React.FC<ColumnListProps> = ({
+    {
+      title: '',
+      field: 'stats',
+      width: 24,
```
Let's make this magic number into a constant.
Maybe `const STATS_COLUMN_WIDTH = 24;`

mgorsk1 replied:
done

Golodhros commented on frontend/amundsen_application/static/js/features/ColumnList/index.tsx:
```diff
@@ -43,6 +43,8 @@ import { logAction } from 'utils/analytics';
 import { buildTableKey, TablePageParams } from 'utils/navigationUtils';
 import { getUniqueValues, filterOutUniqueValues } from 'utils/stats';
 
+import { GraphIcon } from 'components/SVGIcons/GraphIcon';
```
Could you add a couple of extra test cases in 'js/features/ColumnList/index.spec.tsx', in the 'when columns with one usage data entry are passed' and 'when columns with no usage data are passed', looking for this icon to show up?

Golodhros reviewed (approved):
LGTM

Thanks a lot!!
